sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1085,7 +1089,8 @@ public LockRenewalOperation getAutoRenewSessionLock(String sessionId, Duration m
             .flatMap(connection -> connection.getManagementNode(entityPath, entityType))
             .flatMap(serviceBusManagementNode ->
                 serviceBusManagementNode.renewMessageLock(lockToken, getLinkName(null)))
-            .map(instant -> managementNodeLocks.addOrUpdate(lockToken, instant, instant));
+            .map(instant -> managementNodeLocks.addOrUpdate(lockToken, instant,
+                instant.atOffset(ZoneOffset.UTC)).atOffset(ZoneOffset.UTC));
```

Comment:
Is this logic correct? I don't remember making such large transformations like this.

Reply:
You can change the LockContainer to take OffsetDateTime so you don't have to do all these method invocations.

Reply:
I think it's correct. It's just conversion back and forth. 
Will refactor it with issue #14691